Multi gpu support for distributed inference

amechambliss13 opened the pull request:
Implemented support for distributed inference experiments across multiple GPUs with support for prediction length and # GPUs inputs fed through the command line. Also separated distributed inference experiments into dist_inf_base.py and removed from base_script.py

Commit:
Added some preliminary infrastructure and documentation for distributed inference across GPUs

Commit:
Reworked the inference function in proj-utils.py to handle distributed inference in distributed-inference.py

Commit:
Fixed indentation errors

Commit:
Fixed numpy import and ensemble index distribution bugs

Commit:
Improved functionality of inference_ensemble() and debugged

Commit:
Resolved all errors in distributed_inference.py and built in support for command line inputs of # GPUs and # of autoregressive steps. Also implemented distributed inference experiments in dist_inf_base.py and removed distributed inference from base_script.py

magpowell commented on distributed_inference.py:
```diff
@@ -0,0 +1,134 @@
+from diffusers import DiffusionPipeline
+from accelerate import Accelerator
```
what is the difference between distributed-inference.py and distributed_inference.py? can we use just one?

magpowell replied:
I see this was fixed in most recent commit, thx!

amechambliss13 committed:
Deleted unused files

Deleted an old version of the distributed inference implementation and removed outdated distributed_utils.py with old distributed inference implementation

magpowell commented on base_script.py:
```diff
@@ -17,17 +18,19 @@
 from constants import VARIABLES
 from proj_utils import load_model, inference, lat, latitude_weighting_factor, weighted_rmse_channels
-from distributed_utils import inference_ensemble
+from quantize import replace_linear_with_target_and_quantize, W8A16LinearLayer, model_size
```
why are there quantization diffs here? shouldn't these already be here after jakes PR?

magpowell commented on dist_inf_base.py:
```diff
@@ -0,0 +1,151 @@
+prediction_length = sys.argv[1]
+
+# Run the ensemble inference and measure the performance
+inference_results = inference_ensemble(ensemble_init, model, prediction_length, idx_vis, params, device = device,img_shape_x = img_shape_x, img_shape_y = img_shape_y, std = std, m =m, field = field)
```
can we delete this file?
